app/src/main/java/com/example/atheneum/viewmodels/FirebaseRefUtils/DatabaseWriteHelper.java

```diff
@@ -221,43 +222,65 @@ public void onComplete(@Nullable DatabaseError databaseError, @NonNull DatabaseR
      * @param notification
      */
     public static void declineAllRequests(final String bookID, final Notification notification) {
+        Log.i(TAG, "declineAllRequests!");
         DatabaseReference bookRequestRef = RequestCollectionRefUtils
                 .getBookRequestCollectionRef(bookID);
-        bookRequestRef.addChildEventListener(new ChildEventListener() {
-            @Override
-            public void onChildAdded(@NonNull DataSnapshot dataSnapshot, @Nullable String s) {
-                final String requesterID = dataSnapshot.getKey();
-                Log.i(TAG, "requester being declined from book acceptance: " + requesterID);
-                notification.setNotificationReceiverID(requesterID);
-                notification.setRequesterID(requesterID);
-                Log.i(TAG, "notification requester ID: " + notification.getRequesterID());
-                declineRequest(requesterID, bookID, notification, false);
-            }
-
-            @Override
-            public void onChildChanged(@NonNull DataSnapshot dataSnapshot, @Nullable String s) {
-
-            }
-
-            @Override
-            public void onChildRemoved(@NonNull DataSnapshot dataSnapshot) {
-
-            }
-
+        bookRequestRef.addListenerForSingleValueEvent(new ValueEventListener() {

```

Comment:
I changed this to a single value listener instead of a child event listener since the child event listener is still active even after all the requests have been declined. So I decided to instead just iterate over all the requests and delete them using a multi-path update at the end. This way all the requests are deleted but the listener is not active after the requests are deleted.